### PR TITLE
Feature/googlepay

### PR DESCRIPTION
**Pull request by GonzaloTieri**

### CODE REVIEW

File: assets/admin/js/globalpayments-enforce-single-gateway.js [hunk 1]
```diff
@@ -3,20 +3,29 @@
         // Toggle GlobalPayments gateway on/off.
         $( '[data-gateway_id^="globalpayments_"]' ).on( 'click', '.wc-payment-gateway-method-toggle-enabled', function() {
             var toggle = true;
+console.log( $( this ).closest('tr').data('gateway_id')    );
+
```
Comment (apetrovici) [inline assets/admin/js/globalpayments-enforce-single-gateway.js:7]: Please remove this log.

File: src/Gateways/AbstractGateway.php [hunk 2]
```diff
@@ -54,6 +54,9 @@ abstract class AbstractGateway extends WC_Payment_Gateway_Cc {
 	const TXN_TYPE_CHECK_ENROLLMENT        = 'checkEnrollment';
 	const TXN_TYPE_INITIATE_AUTHENTICATION = 'initiateAuthentication';
 
+	// DigitalWallet
+	const DIGITAL_WALLETS = ['globalpayments_googlepay','globalpayments_applepay'];
+
```
Comment (apetrovici) [inline src/Gateways/AbstractGateway.php:59]: const DIGITAL_WALLETS = array(
    GooglePayGateway::GATEWAY_ID
);

File: src/Gateways/GooglePayGateway.php [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+				'type'        => 'text',
+				'description' => __( 'This controls the title which the user sees during checkout.', 'globalpayments-gateway-provider-for-woocommerce' ),
+				'default'     => __( 'Credit Card', 'globalpayments-gateway-provider-for-woocommerce' ),
+				'desc_tip'    => true,
+			),
+		), $this->get_gateway_form_fields() );
+	 }
+
+	public function acceptedCardsOptions()
+	{
```
Comment (apetrovici) [inline src/Gateways/GooglePayGateway.php:226]: Please follow the naming convention accepted_card_options.

File: src/Gateways/GooglePayGateway.php [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+		return array(
+			'VISA' => 'Visa',
+            'MASTERCARD' => 'MasterCard',
+			'AMEX' => 'AMEX',
+			'DISCOVER' =>  'Discover',
+			'JCB' => 'JCB'
+
```
Comment (apetrovici) [inline src/Gateways/GooglePayGateway.php:233]: Alignment and spaces

File: src/Gateways/Helper/Helper.php [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace GlobalPayments\WooCommercePaymentGatewayProvider\Gateways\Helper;
+
+defined( 'ABSPATH' ) || exit;
+
+class Helper {
+
+    public static function removeSlashesFromToken( string $token ): string {
```
Comment (apetrovici) [inline src/Gateways/Helper/Helper.php:9]: Naming convention, alignment and spaces.

**Review by apetrovici — CHANGES_REQUESTED** [review]

**Comment by apetrovici** [comment]
@GonzaloTieri please update the SDK version in composer.json to 3.0.0 and adjust the code where needed.

**Comment by apetrovici** [comment]
@GonzaloTieri there are enough comments for this PR. I will stop reviewing until all of them solved.

1. Please make sure that you don't break the existing functionality. With this code, I wasn't able to perform a transaction with Credit card.
2. Please update the SDK version to at least 3.0.0
3. Make sure you respect the Wordpress coding standards https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/
4. Please remove unused namespaces.
5. Watch out for duplicated code.

File: assets/admin/js/globalpayments-enforce-single-gateway.js (hunk 1 above)
Comment (GonzaloTieri) [reply assets/admin/js/globalpayments-enforce-single-gateway.js:7]: done

File: src/Gateways/GooglePayGateway.php [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+	public $secret_key;
+
+	/**
+	 * Sandbox Merchant location public API key
+	 *
+	 * Used for single-use tokenization on frontend
+	 *
+	 * @var string
+	 */
+	public $sandbox_public_key;
```
[the remark this answers is not in the document]
Comment (GonzaloTieri) [reply src/Gateways/GooglePayGateway.php]: done

File: src/Gateways/GooglePayGateway.php [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+	public $public_key;
+
+	/**
+	 * Live Merchant location secret API key
+	 *
+	 * Used for gateway transactions on backend
+	 *
+	 * @var string
+	 */
+	public $secret_key;
```
[the remark this answers is not in the document]
Comment (GonzaloTieri) [reply src/Gateways/GooglePayGateway.php]: done

File: src/Gateways/GooglePayGateway.php [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+    public $gateway_provider = GatewayProvider::GP_API;
+    
+	/**
+	 * Live Merchant location public API key
+	 *
+	 * Used for single-use tokenization on frontend
+	 *
+	 * @var string
+	 */
+	public $public_key;
```
[the remark this answers is not in the document]
Comment (GonzaloTieri) [reply src/Gateways/GooglePayGateway.php]: done

File: src/Gateways/GooglePayGateway.php [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+use WC_Order;
+use GlobalPayments\Api\Entities\Enums\Environment;
+use GlobalPayments\Api\Entities\Enums\GatewayProvider;
+use GlobalPayments\Api\Entities\MerchantKVP;
+use GlobalPayments\Api\Entities\Enums\GpApi\Channels;
+use GlobalPayments\Api\Entities\Reporting\TransactionSummary;
+use GlobalPayments\Api\Entities\Transaction;
+use GlobalPayments\WooCommercePaymentGatewayProvider\Gateways\HeartlandGiftCards\HeartlandGiftCardOrder;
+use GlobalPayments\WooCommercePaymentGatewayProvider\Plugin;
+
```
[the remark this answers is not in the document]
Comment (GonzaloTieri) [reply src/Gateways/GooglePayGateway.php:10]: done

File: src/Gateways/AbstractGateway.php (hunk 2 above)
Comment (GonzaloTieri) [reply src/Gateways/AbstractGateway.php:59]: done

File: src/Gateways/GooglePayGateway.php (hunk 4 above)
Comment (GonzaloTieri) [reply src/Gateways/GooglePayGateway.php]: done

File: src/Gateways/Helper/Helper.php (hunk 5 above)
Comment (GonzaloTieri) [reply src/Gateways/Helper/Helper.php:9]: done

**Review by apetrovici — APPROVED** [review]

**Review by ahumulescu — APPROVED** [review]

**Pull request merged**